.github/workflows/dotnetcore.yml

```diff
@@ -0,0 +1,41 @@
+name: 'Testing XamlX'
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    strategy:
+      matrix:
+        os: [ windows-latest, ubuntu-latest ]
+
+    name: Build ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+
+    steps:
+
+    - uses: actions/checkout@v2
+    - name: Setup .NET Core
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 3.1.402
+
+    - name: Build TestClasses
+      run: dotnet build tests/TestClasses/TestClasses.csproj
+
+    - name: Test for netcoreapp3.1
+      run: dotnet test --framework netcoreapp3.1
+
+    - name: Test for net47 CecilNetstandardTests
+      if: matrix.os == 'windows-latest'
+      run: dotnet test --framework net47 tests/CecilNetstandardTests/CecilNetstandardTests.csproj
+    - name: Test for net47 CecilTests
+      if: matrix.os == 'windows-latest'
+      run: dotnet test --framework net47 tests/CecilTests/CecilTests.csproj
+    - name: Test for net47 XamlParserTests
+      if: matrix.os == 'windows-latest'
+      run: dotnet test --framework net47 tests/XamlParserTests/XamlParserTests.csproj
```

Comment:
Consider checking out the following GitHub Actions setup:
https://github.com/reactiveui/ReactiveUI.Validation/blob/e7bc86ded6f6eeecc781094ad30283bff68fc5fc/.github/workflows/ci-build.yml#L48-L57

E.g. if you build the project first, you can then run unit tests for all built assemblies ending with `Tests.csproj` using the `*Tests.csproj` glob pattern and the `--no-build` option. Could be useful in removing all these `if` statements and in generating only one report for all the test projects instead.

Reply:
Oh, that looks interesting, thanks!